dozer-types/src/json_types.rs

```diff
@@ -33,13 +80,89 @@ pub fn field_to_json_value(field: Field) -> Value {
         Field::Decimal(n) => Value::String(n.to_string()),
         Field::Timestamp(ts) => Value::String(ts.to_rfc3339_opts(SecondsFormat::Millis, true)),
         Field::Date(n) => Value::String(n.format(DATE_FORMAT).to_string()),
-        Field::Bson(b) => Value::from(b),
+        Field::Json(b) => json_value_to_serde_json(b),
         Field::Point(point) => convert_x_y_to_object(&point.0.x_y()),
         Field::Duration(d) => convert_duration_to_object(&d),
         Field::Null => Value::Null,
     }
 }
 
+pub fn json_value_to_serde_json(value: JsonValue) -> Value {
+    match value {
+        JsonValue::Null => Value::Null,
+        JsonValue::Bool(b) => Value::Bool(b),
+        JsonValue::Number(n) => Value::Number(Number::from_f64(*n).unwrap()),
+        JsonValue::String(s) => Value::String(s),
+        JsonValue::Array(a) => Value::Array(
+            a.iter()
+                .map(|val| json_value_to_serde_json(val.to_owned()))
+                .collect(),
+        ),
+        JsonValue::Object(o) => {
+            let mut values: Map<String, Value> = Map::new();
+            for (key, val) in o {
+                values.insert(key, json_value_to_serde_json(val));
+            }
+            Value::Object(values)
+        }
+    }
+}
+
+pub fn json_value_to_prost_kind(val: JsonValue) -> ProstValue {
+    ProstValue {
+        kind: match val {
+            JsonValue::Null => Some(Kind::NullValue(0)),
+            JsonValue::Bool(b) => Some(Kind::BoolValue(b)),
+            JsonValue::Number(n) => Some(Kind::NumberValue(*n)),
+            JsonValue::String(s) => Some(Kind::StringValue(s)),
+            JsonValue::Array(a) => {
+                let values: prost::alloc::vec::Vec<ProstValue> = a
+                    .iter()
+                    .map(|val| json_value_to_prost_kind(val.to_owned()))
+                    .collect();
+                Some(Kind::ListValue(ListValue { values }))
+            }
+            JsonValue::Object(o) => {
+                let fields: prost::alloc::collections::BTreeMap<
+                    prost::alloc::string::String,
+                    ProstValue,
+                > = o
+                    .iter()
+                    .map(|(key, val)| {
+                        (
+                            prost::alloc::string::String::from(key),
+                            json_value_to_prost_kind(val.to_owned()),
+                        )
+                    })
+                    .collect();
+                Some(Kind::StructValue(Struct { fields }))
+            }
+        },
+    }
+}
+
+// todo: not sure whether we need to involve serde_json conversion, and From<serde_json> for try_get from row
+pub fn serde_json_to_json_value(value: Value) -> JsonValue {
+    match value {
+        Value::Null => JsonValue::Null,
+        Value::Bool(b) => JsonValue::Bool(b),
+        Value::Number(n) => JsonValue::Number(OrderedFloat(n.as_f64().unwrap())),
```

Comment:
Don't `unwrap` here. Return an appropriate `Err` saying that we don't support json numbers out of the `f64` range.